### DISCUSSION TITLE
OBOE CODE SAMPLES

### DISCUSSION THREAD

**Discussion by smokbott** [google/oboe#1646] [Q&A]

**Comment by philburk** [comment]
That code sets the MIDI device, not the audio device. They are very
different. MIDI is for connecting to mechanical peripherals like keyboards.

The microphone is an audio device. You cannot have the same audio device be
an input and an output on Android. Each device is either an input or an
output.

**Comment by smokbott** [comment]
Yes, I really had some confusion, since I'm getting to know the oboe library. In the same way if it is possible to use the same device as input and output: for example an external sound card.
